pkg/runtime/runtime.go

```diff
@@ -1331,138 +1467,58 @@ func (a *DaprRuntime) loadHTTPEndpoints() error {
+	log.Info("Initiating actor shutdown")
 	if a.actor != nil {
 		log.Info("Shutting down actor")
-		a.actor.Stop()
+		return a.actor.Stop()
 	}
+	return nil
 }
 
-func (a *DaprRuntime) stopWorkflow(ctx context.Context) {
+func (a *DaprRuntime) stopWorkflow(ctx context.Context) error {
 	if a.workflowEngine != nil {
 		log.Info("Shutting down workflow engine")
-		a.workflowEngine.Stop(ctx)
+		return a.workflowEngine.Stop(ctx)
 	}
-}
-
-// shutdownOutputComponents allows for a graceful shutdown of all runtime internal operations of components that are not source of more work.
-func (a *DaprRuntime) shutdownOutputComponents(ctx context.Context) error {
-	closers := []concurrency.Runner{func(ctx context.Context) error {
-		log.Info("Shutting down name resolver")
-		if closer, ok := a.nameResolver.(io.Closer); ok && closer != nil {
-			if err := closer.Close(); err != nil {
-				err = fmt.Errorf("error closing name resolver: %w", err)
-				log.Warn(err)
-				return err
-			}
-		}
-		return nil
-	}}
-
-	closeComponent := func(comp componentsV1alpha1.Component) func(context.Context) error {
-		return func(context.Context) error {
-			log.Infof("Shutting down component %s", comp.LogName())
-			if err := a.processor.Close(comp); err != nil {
-				return err
-			}
-			return nil
-		}
-	}
-	for _, comp := range a.compStore.ListComponents() {
-		closers = append(closers, closeComponent(comp))
-	}
-
-	return concurrency.NewRunnerManager(closers...).Run(ctx)
+	return nil
 }
 
 // ShutdownWithWait will gracefully stop runtime and wait outstanding operations.
 func (a *DaprRuntime) ShutdownWithWait() {
-	// Another shutdown signal causes an instant shutdown and interrupts the graceful shutdown
-	go func() {
-		<-ShutdownSignal()
-		log.Warn("Received another shutdown signal - shutting down right away")
-		os.Exit(0)
-	}()
-
-	a.Shutdown(a.runtimeConfig.gracefulShutdownDuration)
-	os.Exit(0)
+	a.close()
+	a.WaitUntilShutdown()
 }
 
-func (a *DaprRuntime) cleanSocket() {
-	if a.runtimeConfig.unixDomainSocket != "" {
-		for _, s := range []string{"http", "grpc"} {
-			os.Remove(fmt.Sprintf("%s/dapr-%s-%s.socket", a.runtimeConfig.unixDomainSocket, a.runtimeConfig.id, s))
-		}
+func (a *DaprRuntime) close() {
+	if a.closing.CompareAndSwap(false, true) {
+		close(a.closeCh)
 	}
 }
 
-func (a *DaprRuntime) Shutdown(duration time.Duration) {
-	// If the shutdown has already been invoked, do nothing
-	if !a.running.CompareAndSwap(true, false) {
-		return
-	}
-
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+func (a *DaprRuntime) WaitUntilShutdown() {
+	<-a.stopped
+}
 
-	if duration > 0 {
-		go func() {
-			select {
-			case <-time.After(duration):
-				log.Fatal("Graceful shutdown timeout exceeded")
-			case <-ctx.Done():
-				return
+func (a *DaprRuntime) cleanSockets() error {
+	var errs []error
+	if a.runtimeConfig.unixDomainSocket != "" {
+		for _, s := range []string{"http", "grpc"} {
```

Comment:
Should remove sockets for pluggable components too? Or not?

Reply:
This is the behavior today and don't think we should do that in this PR.